### ISSUE TITLE
Jira Collector - automatically find field names

### ISSUE THREAD

**Issue by edtan** [Hygieia/Hygieia#3037]
Currently, users need to manually look up `customfieldname`s by examining the results from various Jira REST APIs.  It would be nice if the Jira Collector could lookup these up based on user-provided "friendly" strings (e.g. "sprint", "epic", "story points").

It seems like this might be possible by using the results from the [createmetadata](https://developer.atlassian.com/server/jira/platform/jira-rest-api-example-create-issue-7897248/) endpoint.

I'd be interested in helping contribute this functionality.

**Comment by Sbrenthughes** [comment]
This API doesn't return for out instance of Jira. So I think it is safe to say it wouldn't work for everyone. If you wanted to provide a PR that containing the changes and a flag to use it that would be awesome.

**Comment by edtan** [comment]
Hmm, the comments in `DefaultJiraClient.java` indicate that the it was created against Jira 7.x, and [should work with 5.x and 6.x](https://github.com/Hygieia/Hygieia/blob/master/collectors/feature/jira/src/main/java/com/capitalone/dashboard/collector/DefaultJiraClient.java#L57).  Also, according to the `createmetadata` link above, that API was added in Jira 5.x, which seems quite old so I'm guessing this should cover most users?

**Comment by Sbrenthughes** [comment]
Can you provide a sample api request? Maybe I am just unable to figure out how to use it to pull the custom fields.

**Comment by edtan** [comment]
If you have an account at https://jira.atlassian.com, after logging in, you can make a GET to https://jira.atlassian.com/rest/api/2/issue/createmeta

**Comment by Sbrenthughes** [comment]
Yeah, I see. We don't have access to that API. So a flag is for sure needed.

**Comment by edtan** [comment]
As I started investigating this a bit more, it seems like this other API [rest/api/2/field](https://confluence.atlassian.com/jirakb/how-to-find-id-for-custom-field-s-744522503.html) is better suited for this feature.  Can you check if you have access to this API on your JIRA?  It seems like this API should also be present in [earlier versions of JIRA](https://docs.atlassian.com/software/jira/docs/api/REST/5.1.1/#id343862).

**Comment by Sbrenthughes** [comment]
Can you verify if these match what you see? These are the name key/values for each of the custom fields.
```
jiraSprintDataFieldName -  "name": "Sprint"
jiraEpicIdFieldName -"name": "Epic Link"
jiraStoryPointsFieldName - "name": "Story Points"
jiraTeamFieldName - "name": "Team"
```

**Comment by edtan** [comment]
Sorry for the delay.  Those do indeed match the fields in our instance, except `Team` as we do not have that plugin.  We are also getting a field "type" which I'm planning on using to help with the lookup.  For example, `Sprint` has this type associated with it `com.pyxis.greenhopper.jira:gh-sprint`:

```
{
    "id": "customfield_10100",
    "name": "Sprint",
    "custom": true,
    "orderable": true,
    "navigable": true,
    "searchable": true,
    "clauseNames": [
      "cf[10100]",
      "Sprint"
    ],
    "schema": {
      "type": "array",
      "items": "string",
      "custom": "com.pyxis.greenhopper.jira:gh-sprint",
      "customId": 10100
    }
  }
```
Here are the field types that I'm seeing on our instance - if they match up with what you're seeing, I'll also use these types in the lookup:
`Sprint`: `com.pyxis.greenhopper.jira:gh-sprint`
`Epic Link`: `com.pyxis.greenhopper.jira:gh-epic-link`
`Story Points`: `com.atlassian.jira.plugin.system.customfieldtypes:float`

Also, could you let me know if `Team` has a type associated with it as well so that I can use it in the lookup?

**Comment by Sbrenthughes** [comment]
Interestingly enough, we have 2 teams that pull from this.

**Comment by edtan** [comment]
Oh, these two both correspond with `Team`?  Hmm...  Maybe I'll leave the field type off of the match for Team then, and/or provide an optional parameter to specify the type?
